includes/classes/DistributorPost.php

```diff
@@ -176,6 +183,20 @@ public function __construct( $post ) {
 			$this->connection_direction = 'pulled';
 			$this->connection_id        = (int) get_post_meta( $post->ID, 'dt_original_source_id', true );
 		}
+
+		$this->site_id = get_current_blog_id();
+
+		// Pre-populate the post data to account for switching sites.
```

Comment:
Can you expound on what the problem is? I guess I'm not quite understanding why all the data below needs to be set when we are on a multisite. I'm assuming it has something to do with ensuring we have the right information from the right site but I guess not completely following without being able to see in what situations this comes into play

Reply:
> ensuring we have the right information from the right site

This is the goal.

We we distribute network sites, we run `switch_to_blog()` fairly early on (I'll probably try to delay it) and calling the helper methods was returning either `false` or the wrong post's information if there was a post ID collision.

In the `DistributorPost::post_data()` method, we currently call `get_the_title()` and the `::get_media()`, `::get_terms()` and `::get_meta()` methods. 

I'm wanting to make sure they always return data related to the post and site we set up the object with. 

I guess I need to improve the comment.